scripts/generate_package_diffs_for_flavors/generate_package_diffs_for_flavors.py

```diff
@@ -0,0 +1,290 @@
+    removed = diff_df["Version2"].isnull() & ~diff_df["Version1"].isnull()
+    updated = ~diff_df["Version1"].isnull() & ~diff_df["Version2"].isnull() & (
+            diff_df["Version1"] != diff_df["Version2"])
+    diff_df["Status"] = ""
+    diff_df["Status"].values[new] = "NEW"
+    diff_df["Status"].values[removed] = "REMOVED"
+    diff_df["Status"].values[updated] = "UPDATED"
+    diff_df = diff_df.fillna("")
+    diff_df = diff_df.sort_values("Status", ascending=False)
+    diff_df = diff_df.reset_index(drop=True)
+    return diff_df
+
+
+def convert_requirements_file(package_list_str: str) -> str:
+    def convert_line(line):
+        line = line.replace("|<<<<1>>>>", "|")
+        line = line.replace("==", "|")
+        line = line.replace(">=", "|")
+        line = line.replace("<=", "|")
+        line = re.sub(r"\|$", "", line)
+        if "|" not in line:
+            line += "|<<<<1>>>>"
+        return line
+
+    result = "\n".join(convert_line(line) for line in package_list_str.splitlines())
+    return result
+
+
+def find_package_file_or_alternative(working_copy: Path,
+                                     build_step_path: Path,
+                                     package_list_file_name: str,
+                                     package_list_file_name_alternatives: Dict[str, List[str]]) -> Optional[str]:
+    possible_package_list_file_names = [package_list_file_name]
+    if package_list_file_name in package_list_file_name_alternatives:
+        possible_package_list_file_names += package_list_file_name_alternatives[package_list_file_name]
+    packages_directory = Path(working_copy, build_step_path, "packages")
+    if packages_directory.exists():
+        for package_list_file_name in possible_package_list_file_names:
+            package_list_file_path = Path(packages_directory, package_list_file_name)
+            if package_list_file_path.exists():
+                return package_list_file_name
+    return None
+
+
+def load_package_file_or_alternative(working_copy: Path,
+                                     package_list_file: Path):
+    package_list_str = ""
+    try:
+        package_list_str = parse_package_list_file(Path(working_copy, package_list_file))
+        if package_list_file.name in ["pip3_packages", "pip_packages"]:
+            package_list_str = convert_requirements_file(package_list_str)
+    except Exception as e:
+        print(f"Could not parse {Path(working_copy, package_list_file)}")
+        print(e)
+    return package_list_str
+
+
+def compare_build_step(build_step_path_1: Path, working_copy_1: Path, working_copy_1_name: str,
+                       build_step_path_2: Path, working_copy_2: Path, working_copy_2_name: str) \
+        -> Dict[Tuple[str, Optional[str]], pd.DataFrame]:
+    package_list_file_name_alternatives = {
+        "python3_pip_packages": ["pip3_packages"],
+        "python2_pip_packages": ["pip_packages"]
+    }
```

Comment:
Hm, I thought all files were renamed?

Reply:
Ah, I believe this for the first time, as on previous tags, we still have the old file names.....

Reply:
Correct, this is for the previous release where we had still old names